app/models/filter.rb

```diff
@@ -61,7 +61,12 @@ def self.search_by_limited(key, operator, value)
     { :conditions => conditions }
   end
 
+  # This method attempts to return an existing class that is derived from the resource_type.
+  # In some instances, this may not be a real class (e.g. a typo) or may be nil in the case
+  # of a filter not having been saved yet and thus the permissions objects not being currently
+  # accessible.
   def self.get_resource_class(resource_type)
+    return nil if resource_type.nil?
```

Comment:
If this should only happen during db:seed maybe we want to do ` if !Foreman.in_rake?`? I don't know exactly any other instances where it'd be OK to be in this situation.